src/Hipace.cpp

```diff
@@ -839,86 +846,89 @@ Hipace::ExplicitMGSolveBxBy (const int lev, const int which_slice)
     AMREX_ALWAYS_ASSERT(Comps[which_slice]["Bx"] + 1 == Comps[which_slice]["By"]);
     AMREX_ALWAYS_ASSERT(Comps[which_slice]["Sy"] + 1 == Comps[which_slice]["Sx"]);
 
-    // construct slice geometry
-    // Set the lo and hi of domain and probdomain in the z direction
-    amrex::RealBox tmp_probdom({AMREX_D_DECL(Geom(lev).ProbLo(Direction::x),
-                                             Geom(lev).ProbLo(Direction::y),
-                                             Geom(lev).ProbLo(Direction::z))},
-                               {AMREX_D_DECL(Geom(lev).ProbHi(Direction::x),
-                                             Geom(lev).ProbHi(Direction::y),
-                                             Geom(lev).ProbHi(Direction::z))});
-    amrex::Box tmp_dom = Geom(lev).Domain();
-    const amrex::Real hi = Geom(lev).ProbHi(Direction::z);
-    const amrex::Real lo = hi - Geom(lev).CellSize(Direction::z);
-    tmp_probdom.setLo(Direction::z, lo);
-    tmp_probdom.setHi(Direction::z, hi);
-    tmp_dom.setSmall(Direction::z, 0);
-    tmp_dom.setBig(Direction::z, 0);
-    amrex::Geometry slice_geom = amrex::Geometry(
-        tmp_dom, tmp_probdom, Geom(lev).Coord(), Geom(lev).isPeriodic());
-
-    slice_geom.setPeriodicity({0,0,0});
-
     amrex::MultiFab& slicemf_BxBySySx = m_fields.getSlices(lev, which_slice);
     amrex::MultiFab BxBy (slicemf_BxBySySx, amrex::make_alias, Comps[which_slice]["Bx"], 2);
     amrex::MultiFab SySx (slicemf_BxBySySx, amrex::make_alias, Comps[which_slice]["Sy"], 2);
 
     amrex::MultiFab& slicemf_chi = m_fields.getSlices(lev, which_slice_chi);
     amrex::MultiFab Mult (slicemf_chi, amrex::make_alias, Comps[which_slice_chi]["chi"], ncomp_chi);
 
+    if (lev!=0) {
+        m_fields.SetBoundaryCondition(Geom(), lev, "Bx", islice,
+                                      m_fields.getField(lev, which_slice, "Sy"));
+        m_fields.SetBoundaryCondition(Geom(), lev, "By", islice,
+                                      m_fields.getField(lev, which_slice, "Sx"));
+    }
+
 #ifdef AMREX_USE_LINEAR_SOLVERS
     if (m_use_amrex_mlmg) {
         // Copy chi to chi2
         m_fields.duplicate(lev, which_slice_chi, {"chi2"}, which_slice_chi, {"chi"});
         amrex::Gpu::streamSynchronize();
-        if (!m_mlalaplacian){
+        if (m_mlalaplacian.size()==0) {
```

Comment:
```suggestion
        if (m_mlalaplacian.size()!=maxLevel()+1) {
```
This should be a bit more future-proof (although I agree that currently this size can only have 2 values, 0 and maxLevel()+1, but this would avoid a segfault if this behaviour changes later).

Reply:
What is currently implemented would also work if there are more levels because `m_mlalaplacian` can only be empty or fully initialized (it gets resized to  maxlev+1 not lev+1).